### ISSUE TITLE
Unable to install any python version on MacOS 12.0.1 Macbook Pro 2016: "internal configure error for the platform triplet"

### ISSUE THREAD

**Issue by bluecpp** [pyenv/pyenv#2171]
[ x] Platform information: MacOS Monterey 12.0.1
    [ x] OS architecture (e.g. amd64): Intel x86_64
    [ x] pyenv version: 2.2.2
    [ x] Python version: no version works
    [ x] C Compiler information: Apple clang version 13.0.0 - Target: arm64-apple-darwin21.1.0

I tried to follow other similar issues such as #1737, however, I was unable to fix this problem. I have included the error log down below.
[log.txt](https://github.com/pyenv/pyenv/files/7614051/log.txt)
-------------------------------------------------------------------------------------------
$ pyenv install 3.7.12 # for example
python-build: use openssl@1.1 from homebrew
python-build: use readline from homebrew
Downloading Python-3.7.12.tar.xz...
-> https://www.python.org/ftp/python/3.7.12/Python-3.7.12.tar.xz
Installing Python-3.7.12...
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 12.0.1 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/14/3sdry2dj3tjdmqpqm2z5z34r0000gn/T/python-build.20211128190121.23895
Results logged to /var/folders/14/3sdry2dj3tjdmqpqm2z5z34r0000gn/T/python-build.20211128190121.23895.log

Last 10 log lines:
checking for --with-cxx-main=<compiler>... no
checking for clang++... no
configure:

  By default, distutils will build C++ extension modules with "clang++".
  If this is not intended, then set CXX on the configure command line.
  
checking for the platform triplet based on compiler characteristics... darwin
configure: error: internal configure error for the platform triplet, please file a bug report
make: *** No targets specified and no makefile found.  Stop.

-------------------------------------------------------------------------------------------
Also the $ pyenv doctor command returns: Congratulations! You are ready to build pythons!

**Comment by native-api** [comment]
Could you also attach `config.log` from the build tree? Most probably, 3.7.12 is incompatible with MacOS 12 but the log may give a different reason.

**Comment by bluecpp** [comment]
Thank you for your reply.
It is not only the version 3.7.12. 
I tired installing different versions of python such as 3.10.0, 3.8.12 as well. However, I was not successful. 
The version 3.9.9 is the only one that seems to be working among the versions that I have tried.

Where can I find the config.log file?

**Comment by native-api** [comment]
> Where can I find the config.log file?

> > Inspect or clean up the working tree at /var/folders/14/3sdry2dj3tjdmqpqm2z5z34r0000gn/T/python-build.20211128190121.23895

**Comment by bluecpp** [comment]
Thank you so much. This solved the problem.

**Comment by genewliu** [comment]
I have a similar problem
[config.log](https://github.com/pyenv/pyenv/files/8102386/config.log)
. Can I resolve it by deleting Homebrew's LLVM? If so, can there be a step by step approach please?

**Comment by amirnn** [comment]
You do not need to uninstall the brew LLVM, just make sure it is not included in your PATH enviornment varaible.
You can do so by printing the value of the $PATH. Issue the command on your command line:
```shell
echo $PATH
```
If you see the "/usr/local/opt/llvm/bin" or a similar value pointing to an LLVM installation, find out where it gets added to your PATH variable, it most definitly gets added by the user's configuration for the the current shell you are using.

For example, if you are using BASH go to your HOME directory.
```shell
cd ~
```
Once, you are there edit the file ".bashrc". In case you are using another shell emulator find the respective file for yours and edit that (zsh -> .zshrc, etc.). Once, you have opened you config file, find the line that adds that value. It would probbly be something like:

```shell
export PATH="/usr/local/opt/llvm/bin:$PATH"
```
Comment that line meaning add # to the beginning, so that zsh or bash would skip that command and would treat it as comment. This way the brew's LLVM would not get inclueded in youtr PATH variable.

**Comment by eric-sawyer-vifive** [comment]
I had this exact error (using Monterey 12.2), and also could only install 3.9.9, no other version. After making sure LLVM was not on my path, reinstalling Xcode, reinstalling pyenv, and reinstalling Xcode command line tools, what finally worked was 
`sudo xcode-select --reset`

**Comment by genewliu** [comment]
> You do not need to uninstall the brew LLVM, just make sure it is not included in your PATH enviornment varaible. You can do so by printing the value of the $PATH. Issue the command on your command line:
> 
> ```shell
> echo $PATH
> ```
> 
> If you see the "/usr/local/opt/llvm/bin" or a similar value pointing to an LLVM installation, find out where it gets added to your PATH variable, it most definitly gets added by the user's configuration for the the current shell you are using.
> 
> For example, if you are using BASH go to your HOME directory.
> 
> ```shell
> cd ~
> ```
> 
> Once, you are there edit the file ".bashrc". In case you are using another shell emulator find the respective file for yours (zsh -> .zshrc, etc.). Once, you found and you have opened your config file, find the line that adds that value. It would probbly be something like:
> 
> ```shell
> export PATH="/usr/local/opt/llvm/bin:$PATH"
> ```
> 
> Comment that line meaning add # to the beginning, so that zsh or bash would skip that command and would treat it as a comment. This way the brew's LLVM would not get inclueded in your PATH variable.
> 
> Another thing you can do is to issue the command:
> 
> ```shell
> pyenv doctor
> ```
> 
> And see if that results in any errors, if so search that specific error.

When I types $PATH, it gives this: zsh: no such file or directory: /Users/geneliu/.pyenv/shims:/Users/geneliu/.pyenv/shims:/Users/geneliu/.pyenv/shims:/Users/geneliu/.pyenv/bin:/Users/geneliu/.pyenv/shims:/Users/geneliu/.pyenv/shims:/Users/geneliu/.pyenv/shims:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/geneliu/documents/webdriver:/Library/Apple/usr/bin:/Users/geneliu/.pyenv/bin:/usr/local/anaconda3/bin:/usr/local/anaconda3/condabin:/Users/geneliu/.pyenv/shims

I cannot find any LLVM for deletion.

**Comment by genewliu** [comment]
> I had this exact error (using Monterey 12.2), and also could only install 3.9.9, no other version. After making sure LLVM was not on my path, reinstalling Xcode, reinstalling pyenv, and reinstalling Xcode command line tools, what finally worked was
> `sudo xcode-select --reset`

Did you use Homebrew to install pyenv or Github Checkout?

**Comment by eric-sawyer-vifive** [comment]
> Did you use Homebrew to install pyenv or Github Checkout?

Most recently I reinstalled with Github Checkout, but before I upgraded to Monterey I had used Homebrew in the past. I'm not sure what ended up causing the problem, but maybe someone else can shed more light.

**Comment by genewliu** [comment]
> 

I tried to reinstall pyenv by Git Checkout. Same BUILD FAILED problem persists just like when I used Homebrew to install pyenv.

BUILD FAILED does not happen to all Python versions. I successfully installed 3.9.10, 3.9.8 by pyenv install.

Therefore, the problem seems to lie with "pyenv install" relating to a bunch of Python versions.

**Comment by native-api** [comment]
@genewliu Your problem is a duplicate of https://github.com/pyenv/pyenv/issues/1737. 3.6 is incompatible with MacOS 11+.

**Comment by genewliu** [comment]
I tried other versions of 3.7.8, 3.8.10, both resulted in the same error in https://github.com/pyenv/pyenv/issues/2171#issuecomment-1046036667

**Comment by ryanrolds** [comment]
Fresh M1 MBP w/ macOS 12.0.1. Can install 3.9.9, but 3.9.1 gives me:
```
checking for the platform triplet based on compiler characteristics... darwin
configure: error: internal configure error for the platform triplet, please file a bug report
```

`PATH` appears to be fine and I've run `xcode-select --reset` with no affect.

**Comment by dhalperi** [comment]
Same setup as @cristifalcas - was working fine yesterday, after last night's auto-upgrading to macOS 12.3 and new cli tools, does not work. I've attached relevant logs:
[config.log.gz](https://github.com/pyenv/pyenv/files/8255805/config.log.gz)
[python-build.20220315102018.49150.log.gz](https://github.com/pyenv/pyenv/files/8255816/python-build.20220315102018.49150.log.gz)

**Comment by mcouthon** [comment]
[Cross posting](https://github.com/pyenv/pyenv/issues/2143#issuecomment-1069223994) here what worked for me.

**Comment by khanmoazzam** [comment]
Having the same issue on my macOS 12.3 with M1 chip. Have tried to install older versions of python either using pyenv or from sources nothing works. Have tried all the solutions that have been suggested above and many more. Please do share if you find a solution.

**Comment by joshteng** [comment]
I managed to get Python 3.9.2 installed MacOS 12.3.1 with M1

@khanmoazzam 
Try this patch https://github.com/python/cpython/commit/9c47667.patch?full_index=1

I use asdf-python to manage various Python versions.

```
export ASDF_PYTHON_PATCH_URL="https://github.com/python/cpython/commit/9c47667.patch?full_index=1"
asdf install python 3.9.2
```

I've not used Pyenv in awhile. Hopefully the first answer is helpful https://stackoverflow.com/questions/65349982/issues-installing-python-3-x-via-pyenv

Reference
https://github.com/pyenv/pyenv/issues/2143#issuecomment-1072032647
https://github.com/python/cpython/commit/9c4766772cda67648184f8ddba546a5fc0167f91

**Comment by Wellers0n** [comment]
> Python released a 3.7.13 version today , and patch versions for several other minor versions. This unblocks it for me, once pyenv has support for them.

Great this worked for me
